File: frontend/src/app/models/o-auth/OAuthPagedResponse.ts
import { AggregatedAppDto } from './AggregatedAppDto';

export type OAuthPagedResponse = {
  apps: AggregatedAppDto[];
  nextPageToken: string;
  allFilteredApps: number;
  allHighRiskApps: number;
  allNotHighRiskApps: number;
};

File: api/src/main/java/com/cloudmen/cloudguard/dto/oAuth/OAuthPagedResponse.java
package com.cloudmen.cloudguard.dto.oAuth;

import java.util.List;

public record OAuthPagedResponse(
        List<AggregatedAppDto> apps,
        String nextPageToken,
        int allFilteredApps,
        int allHighRiskApps,
        int allNotHighRiskApps
) {
}
